plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerConnectorTest.java

```diff
@@ -662,6 +664,229 @@ protected void verifyColumnNameLengthFailurePermissible(Throwable e)
+                    "The statement did not return a result set.");
+        }
+    }
+
+    @Test
+    public void testProcedureWithIfElseStatement()
+    {
+        try (TestProcedure testProcedure = createTestingProcedure(
+                "@id INTEGER",
+                """
+                IF @id > 50
+                    SELECT 1 as first_column;
+                ELSE
+                    SELECT '2' as second_column;
+                """)) {
+            assertQueryFails(
+                    "SELECT * FROM TABLE(system.procedure(query => 'EXECUTE %s.%s 100')) "
+                            .formatted(getSession().getSchema().orElseThrow(), testProcedure.getName()),
+                    "Procedure has multiple ResultSets for query: .*");
+        }
+    }
+
+    @Test
+    public void testProcedureWithMultipleResultSet()
+    {
+        try (TestProcedure testProcedure = createTestingProcedure("SELECT 1 as first_row; SELECT 2 as second_row")) {
+            assertQueryFails(
+                    "SELECT * FROM TABLE(system.procedure(query => 'EXECUTE %s.%s')) "
+                            .formatted(getSession().getSchema().orElseThrow(), testProcedure.getName()),
+                    "Procedure has multiple ResultSets for query: .*");
+        }
+    }
+
+    @Test
+    public void testProcedureWithCreateOperation()
+    {
+        String tableName = "table_to_create" + randomNameSuffix();
+        try (TestProcedure testProcedure = createTestingProcedure("CREATE TABLE %s (id BIGINT)".formatted(tableName))) {
+            assertQueryFails(
+                    "SELECT * FROM TABLE(system.procedure(query => 'EXECUTE %s.%s'))"
+                            .formatted(getSession().getSchema().orElseThrow(), testProcedure.getName()),
+                    "Failed to get table handle for procedure query. The statement did not return a result set.");
+            assertQueryReturnsEmptyResult("SHOW TABLES LIKE '%s'".formatted(tableName));
+        }
+    }
+
+    @Test
+    public void testProcedureWithDropOperation()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "table_to_drop", "(id BIGINT)");
+                TestProcedure testProcedure = createTestingProcedure("DROP TABLE " + table.getName())) {
+            assertQueryFails(
+                    "SELECT * FROM TABLE(system.procedure(query => 'EXECUTE %s.%s'))"
+                            .formatted(getSession().getSchema().orElseThrow(), testProcedure.getName()),
+                    "Failed to get table handle for procedure query. The statement did not return a result set.");
+            assertQuery("SHOW TABLES LIKE '%s'".formatted(table.getName()), "VALUES '%s'".formatted(table.getName()));
+        }
+    }
+
+    @Test
+    public void testProcedureWithInsertOperation()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "table_to_insert", "(id BIGINT)");
```

Comment:
I'm confused. You stated in PR description that non-select statements are not supported. And here we have insert, delete and update...

Reply:
We have a test that fails for these operations and also check if doesn't modify the underlying data.

Reply:
I totally missed the fact that these are negative assertions.